WebDriverAgentLib/Commands/FBAlertViewCommands.m

```diff
@@ -44,7 +45,13 @@ + (NSArray *)routes
 + (id<FBResponsePayload>)handleAlertAcceptCommand:(FBRouteRequest *)request
 {
   FBSession *session = request.session;
-  if (![[FBAlert alertWithApplication:session.application] acceptWithError:nil]) {
+  NSString *name = request.parameters[@"name"];
+
+  if ([name length] != 0) {
+    if (![[FBAlert alertWithApplication:session.application] clickAlertButton:name error:nil]) {
+      return FBResponseWithStatus(FBCommandStatusNoAlertPresent, nil);
```

Comment:
I would suggest:
1) Checking if alert is present, and raise `FBCommandStatusNoAlertPresent` if not,
2) Dismiss/Accept with/without button specified and raise `FBResponseWithErrorFormat` in case of failure.